### ISSUE TITLE
Parsing error with C++20 requires clause and fully qualified result type

### ISSUE THREAD

**Issue by seijikun** [doxygen/doxygen#8980]
**Describe the bug**
Doxygen incorrectly parses the requires clauses of the following snippet:
```c++
template<typename T>
struct IterApi {
	
	namespace ItemOwned = int;
	
	template<typename TResult = ItemOwned>
	requires requires(TResult res, ItemOwned item) {
		{ item < res };
		{ res = item };
	}
	std::optional<TResult> min() {
		auto item = Iterator::next(*self());
		if(!item.hasValue()) { return {}; }
		TResult result = item.value();
		forEach([&result](Item&& item) {
			if(item < result) { result = item; }
		});
		return result;
	}
	
}:
```

**Screenshots**
This is parsed to the following declaration, where return value and function name are incorrectly parsed into the template and requires clause:
![image](https://user-images.githubusercontent.com/820845/147269915-d0c018bc-c28d-4fce-88e7-bdbe8ebd8bff.png)

**To Reproduce**
[requires-bug.tar.gz](https://github.com/doxygen/doxygen/files/7770537/requires-bug.tar.gz)

**Version**
Both `1.9.2` and `c94a1ee83db4eb098e72db3fe992ff4b8b9ace56`.

**Context**
When changing the method's result value from `std::optional<TResult>` to `TResult`, it parses properly.

**Comment by albert-github** [comment]
I think the problem here is the `requires` clause analogous to the problem mentioned in issue #8911: Undocumented function with C++ requires clause which has been fixed by means of commit e55dae5c9bca0a95934fe7c3c6dbba9c63142c92 in the current master.

The output with the current master (1.9.3 (c94a1ee83db4eb098e72db3fe992ff4b8b9ace56)) looks like:

![image](https://user-images.githubusercontent.com/5223533/147271583-7842c196-fa13-4eb1-97c9-93bd567fa2c8.png)

I think the problem is a duplicate and fixed but not released

**Comment by seijikun** [comment]
Yes, you are right!
Apparently, deploying to `gh-pages` takes some while, and it actually looks fine with `c94a1ee83db4eb098e72db3fe992ff4b8b9ace56`.
Sorry!
